workspaces/templates-lib/packages/infra-aws/src/awsUserUtils.ts

```diff
@@ -0,0 +1,193 @@
+    'AWS_SECRET_ACCESS_KEY not defined'
+  );
+  const region = process.env.AWS_REGION || process.env.AWS_DEFAULT_REGION;
+  assert(region, 'Neither AWS_REGION nor AWS_DEFAULT_REGION are defined.');
+  const credentials = new AWS.EnvironmentCredentials('AWS');
+  await credentials.getPromise();
+  AWS.config.credentials = credentials;
+
+  AWS.config.update({ region });
+
+  return credentials;
+}
+/**
+ * Obtains AWS user credentials from container environment variables for ECS containers.
+ */
+export async function getAWSUserFromContainerEnvironment(): Promise<AWS.ECSCredentials> {
+  const ecsCredentials = new AWS.ECSCredentials({
+    httpOptions: { timeout: 5000 },
+    maxRetries: 10, // retry 10 times
+  });
+  await ecsCredentials.getPromise();
+
+  AWS.config.credentials = ecsCredentials;
+
+  if (!process.env.AWS_REGION) {
+    throw new Error(
+      'AWS region environment variable ("AWS_REGION") not defined for ECS task.'
+    );
+  }
+
+  AWS.config.update({ region: process.env.AWS_REGION });
+
+  return ecsCredentials;
+}
+
+export async function getAWSUserFromDefaultLocalProfile(): Promise<AWS.Credentials> {
+  let credentials = new AWS.SharedIniFileCredentials();
+
+  // if no access key is found, try loading process_credentials
+  if (!credentials.accessKeyId) {
+    credentials = new AWS.ProcessCredentials();
+    await credentials.refreshPromise();
+  }
+
+  AWS.config.credentials = credentials;
+  // see https://github.com/aws/aws-sdk-js/pull/1391
+  process.env.AWS_SDK_LOAD_CONFIG = 'true';
+  return credentials;
+}
+
+export async function getAWSUserFromGoldstackConfig(
+  config: AWSConfiguration,
+  userName: string
+): Promise<AWS.Credentials> {
+  const user = config.users.find((user) => user.name === userName);
+  if (!user) {
+    throw new Error(`User '${userName}' does not exist in AWS configuration.`);
+  }
+
+  if (user.type === 'profile') {
+    const userConfig = user.config as AWSProfileConfig;
+
+    if (process.env.AWS_SHARED_CREDENTIALS_FILE) {
+      console.warn(
+        `Using AWS_SHARED_CREDENTIALS_FILE environment variable: '${process.env.AWS_SHARED_CREDENTIALS_FILE}'. awsCredentialsFileName in configuration will be ignored.`
+      );
+    }
+
+    const envVarValues = {
+      AWS_SDK_LOAD_CONFIG: process.env.AWS_SDK_LOAD_CONFIG,
+      AWS_SHARED_CREDENTIALS_FILE: process.env.AWS_SHARED_CREDENTIALS_FILE,
+      AWS_CONFIG_FILE: process.env.AWS_CONFIG_FILE,
+    };
+    if (userConfig.awsConfigFileName) {
+      // support loading from both `config` and `credentials` files, see https://github.com/goldstack/goldstack/issues/17#issuecomment-1044811805
+      process.env.AWS_SDK_LOAD_CONFIG = '1';
+      // filename property is ignored if AWS_SDK_LOAD_CONFIG is set; thus need to set AWS_SHARED_CREDENTIALS_FILE.
+      process.env.AWS_SHARED_CREDENTIALS_FILE =
+        userConfig.awsCredentialsFileName;
+      process.env.AWS_CONFIG_FILE = userConfig.awsConfigFileName;
+    }
```

Comment:
This block won't get called if `awsConfigFileName` isn't provided, and so it won't support getting a profile from the default location `~/.aws/config` without tweaks.

Reply:
Changed this now so that it is always set - I don't think that would do any harm? https://github.com/goldstack/goldstack/pull/102/files#diff-e43f64484d679c0de71cca9f787c4dc4d20e104fbd0faada8b38efc275670918R91

Reply:
I guess it *shouldn't* do any harm, I was just being cautious.

You have more faith in AWS Credentials internals than me :)

Reply:
In fact, it did do harm and prevented the loading of the credentials from the `credentials` file. So changed it now that this variable is only set when we want to load from the `config` file!